bin/hrx

```diff
@@ -34,12 +41,74 @@ class HRX::CLI < Thor
     raise Thor::Error.new(e.message)
   end
 
+  desc "extract <archive>", "Extract files from an archive"
+  option :directory, aliases: :d, type: :string,
+         banner: "Directory to which files are extracted"
+  long_desc <<-LONGDESC
+    Extract the files from an archive.
+
+    This puts the files in the directory given by the --directory option, which
+    defaults to the name of the archive without ".hrx".
+
+    > hrx extract test/specs.hrx
+
+    Extracted 12 files to test/specs.
+  LONGDESC
+  def extract(archive)
+    directory = options[:directory] || begin
+        unless archive.end_with?(".hrx")
+          raise Thor::Error.new(
+            "Can't determine a default output directory for #{archive}. Pass one " +
+            "with --directory.")
+        end
+        directory = archive[0...-".hrx".length]
+      end
+
+    hrx = HRX::Archive.load(archive)
+    files_extracted = hrx.entries.count do |entry|
+      destination = File.join(directory, entry.path)
+      if entry.is_a?(HRX::Directory)
+        FileUtils.mkdir_p(destination)
+        false
+      else
+        FileUtils.mkdir_p(File.dirname(destination))
+        if !File.exist?(destination) ||
+           file_collision(destination) {entry.content}
+          File.write(destination, entry.content)
+          true
+        else
+          false
+        end
+      end
+    end
+
+    $stderr.puts(shell.set_color(
+      "Extracted #{_numberOf(files_extracted, 'file')} to #{directory}.",
+      :green))
+  rescue HRX::Error, SystemCallError => e
+    raise Thor::Error.new(e.message)
+  end
+
   def self.exit_on_failure?
     true
   end
 
   private
 
+  # Returns a description of the number of `singular`s.
+  #
+  # `plural` defaults to `singular` with a trailing "s".
+  def _numberOf(number, singular, plural: nil)
```

Comment:
It looks like `plural` will never not be nil, so it might be unnecessary to have it as optional function args to `_numberOf` and `_pluralize`

Reply:
I'd like to keep it around since it's fairly likely we'll need to pluralize some irregular words at some point.

Reply:
👍